File: src/api/provider-limits.ts
import {
	type ProviderLimitsResponse,
	provider_limits_response_schema,
} from "@contracts/provider-limits/snapshots";
import { commands } from "@/platform/ipc";

export async function get_provider_limits(): Promise<ProviderLimitsResponse> {
	const raw = await commands.provider_limits.get_provider_limits();
	return provider_limits_response_schema.parse(raw);
}

export async function refresh_provider_limits(): Promise<ProviderLimitsResponse> {
	const raw = await commands.provider_limits.refresh_provider_limits();
	return provider_limits_response_schema.parse(raw);
}

File: src-tauri/src/commands/provider_limits.rs
use tauri::State;
use crate::models::provider_limits::ProviderLimitsResponse;
use crate::provider_limits::cache::ProviderLimitsCache;

/// Return cached provider limits, fetching on first call.
#[tauri::command]
pub async fn get_provider_limits(
    cache: State<'_, ProviderLimitsCache>,
) -> Result<ProviderLimitsResponse, String> {
    Ok(cache.get().await)
}

/// Force-refresh all provider limits and return fresh data.
#[tauri::command]
pub async fn refresh_provider_limits(
    cache: State<'_, ProviderLimitsCache>,
) -> Result<ProviderLimitsResponse, String> {
    Ok(cache.refresh().await)
}
